traversal/walk.go

```diff
@@ -120,6 +120,25 @@ func (prog Progress) walkAdv(n datamodel.Node, s selector.Selector, fn AdvVisitF
 }
 
 func (prog Progress) walkAdv_iterateAll(n datamodel.Node, s selector.Selector, fn AdvVisitFn) error {
+	if rs, ok := s.(selector.Transformable); ok {
+		adl := rs.Transform()
+		if prog.Cfg.LinkSystem.KnownReifiers == nil {
+			return fmt.Errorf("adl requested but not supported by link system: %s", adl)
+		}
+		reifier, ok := prog.Cfg.LinkSystem.KnownReifiers[adl]
+		if !ok {
+			return fmt.Errorf("unregistered adl requested: %s", adl)
```

Comment:
```suggestion
			return fmt.Errorf("unregistered adl requested: %q", adl)
```

Reply:
%q is "a single-quoted character literal " - i don't think the adl is guaranteed to be a single character?

Reply:
It has different meanings depending on the argument type. For strings:

> %q	a double-quoted string safely escaped with Go syntax